debian/wviewcleardata.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH WVIEWCLEARDATA 1 "January 24, 2010"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
wviewcleardata \- script to purge all wview weather data
.SH SYNOPSIS
.B wviewcleardata
.RI
.br
.SH DESCRIPTION
This manual page describes the use of
.B wviewcleardata
.
Purge all weather data stored by a wview installation.
.BR
.BR Note: wview should not be running when running this script.
.BR Note: must be executed on the source wview server.
.SH OPTIONS
There are no options.
.TP
.SH SEE ALSO
.BR wviewconfig (1),
.BR /usr/share/doc/wview/wview-User-Manual.html
.br
.SH AUTHOR
wviewcleardata was written by Mark Teel.
.PP
This manual page was written by Mark Teel <mteel2005@gmail.com>,
for the Debian project (and may be used by others).


ChangeLog
RELEASE VERSION HISTORY (newest to oldest)
------------------------------------------


5.21.5    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Remove debug message.




5.21.4    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Added stale pid file cleanup to the APT package start script.

2)  WMRUSB HID interface changes. Changed reader thread to blocking IO.
    Mitigated the console occasionally blasting out bogus data. Now zero
    packet loss. Fixed issue with pressure sanity check. Added sanity
    check for bogus rain accumulator values from the WMR console.




5.21.3    03-31-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Further WMRUSB interface improvements. Better packet framing.




5.21.2    03-30-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Updated debian APT postinst script to include copying the exfoliation skin
    for new installs.

2)  Enabled checksum processing for WMR stations.

3)  Changed LOOP windspeed items to floats for better precision.
    NOTE: If you use custom datafeed clients the LOOP data structure has changed.

4)  Added temp only packet support for WMRUSB stations. This adds support for
    add-on sensors such as the pool sensor.

5)  Fixed APT installs changing the wview.log file ownership incorrectly. Also
    restarts rsyslog automatically if a new install.




5.21.1    03-13-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Modified HILOW database storage so higher precision is used for cumulative
    data such as rain and ET.

2)  Fixed issue with yearly historical charts when generated at wview start
    (as opposed to at midnight). The current day was erroneously being included in
    the history data. This was most apparent with the UV and Solar Radiation
    charts.

3)  Fixed bug introduced in 5.21.0 release causing NULL wind direction to have
    360 added to it (to normalize calibration result). A value of -99640 was
    being stored/displayed.




5.21.0    03-05-2014
--------------------

radlib version required: 2.12.0 or newer

1)  WMRUSB improvements. Further reverse engineering of the undocumented protocol. Reset and
    heartbeat messages now working. Packet reception more stable. Initial packet acquisition
    faster resulting in faster startup.

2)  Modify CWOP packet header for new stricter compliance requirements of CWOP servers.

3)  Fix log message time_t bug for 64-bit platforms.

4)  Add timeout parameter for ssh rules. Added rows to config database.

5)  Updated data feed conversion logic to handle ARCHIVE_VALUE_NULL in loop data.

6)  Added flat line sensor value detection for outside temp, wind direction and wind speed.
    If there are no changes over a 4 hour period an email alert will be sent.

7)  Added Station Name, City and State to the wviewmgmt System Status page header.

8)  Added logic to SSH timer to fixup next timer interval if system time has skewed or
    if the ssh transfer took longer than 2 minutes.

9)  Fixed TE923 rain calculation to assume each tip is 1/36" instead of 1mm.

10) Fixed a bug for Vantage Pro stations which could cause the last archive interval of
    the previous day to be included in the current day HILOW data. This occurred if the
    midnight archive record was provided late by the console.

11) Added wet bulb temperature calculation and html tag --WetBulbTemp--. Added to all
    "readings.incx" templates.

12) Added email alert notification if no loop data has been received for a given archive
    interval. This applies to all stations except the Davis stations (alerts already
    exist for them).

13) Updated start scripts to detect and delete stale pid files.

14) Modified WH1080 data parsing to assume a humidity of 9% if the value from the
    station is undefined (0xff).




5.20.2    03-21-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added back "dailyrainin" for http reporting. They do no computation based on hourly rain values.

2)  Fixed a typo in the wviewmgmt station.php file. 




5.20.1    03-20-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Modified http rain submission (Wunderground and Weatherforyou) so only the
    "rainin" parameter is sent so daily rain will be correct. 




5.20.0    03-17-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added "wait_for_time_set" function to most wview start script examples so 
    wview daemons are not started until the system time is set. This applies to
    platforms without a real time clock which must reset their time during 
    startup.

2)  Updated the User Manual section on skin selection to better describe enabling
    extended data generation (UV, ET, radiation).

3)  Updated the debian post install script to try harder to find the default http
    document root location for creation of symbolic links to wviewmgmt and the
    weather home page.

4)  Added new configuration option to optionally disable using the Vantage Pro 
    console archive records and instead autogenerate them from the LOOP data.
    This is how all other stations work. It also supports non-standard console
    interfaces (those not using the datalogger on the console).

5)  Updated the Weatherforyou URL for data submissions.

6)  Fixed bug with DataFeed clients that caused an infinite loop if you
    restarted wview while there was a datafeed client connection.

7)  Fixed bug with Alarm Type configuration introduced in 5.19.0.

8)  Changed TWI station to compute rain fall rate as it is reported to not
    be provided by the station.

9)  Added socket retry logic to the station ethernet interface (used by the
    Vantage Pro IP datalogger and any serial-to-ethernet device).

10) Modified CWOP host name to IP address resolution so that 'getaddrinfo' is
    used instead of 'gethostbyname' and all results are tried in the order they 

wviewconfig/wviewcleardata.sh
################################################################################
#
# File:           wviewcleardata.sh
#
# Description:    Provide a script to remove all weather data from the databases
#
# Usage:          (must be root)
#                 wviewcleardata
#
# History:
# Engineer	  Date	    Ver   Comments
# MS Teel	  01/24/10   1    Original
#
################################################################################


################################################################################
#################################  M A C R O S  ################################
################################################################################
WVIEWD_PID=$WVIEW_DATA_DIR/wviewd.pid

################################################################################
#######################  D E F I N E  F U N C T I O N S  #######################
################################################################################

show_usage()
{
    echo ""
    echo "wviewcleardata"
    echo "    removes all weather data from the databases"
    echo ""
}


interactive_intro()
{
    echo "################################################################################"
    echo " !!!!!!!!!!!!!!!!          READ THIS BEFORE PROCEEDING         !!!!!!!!!!!!!!!!"
    echo ""
    echo "--> wview database purging utility"
    echo ""
    echo "--> Note: This script will REMOVE all archive records and DELETE the following"
    echo "          databases:"
    echo "          wview-hilow.sdb (HILOW data)"
    echo "          wview-history.sdb (daily history)"
    echo "          wview-noaa.sdb (NOAA data)"
    echo ""
    echo "--> Use this script to reset all data - i.e., remove simulator data to begin"
    echo "    using your real station data."
    echo ""
    echo "--> It is highly recommended that you BACKUP YOUR ARCHIVE DIRECTORY before doing"
    echo "    this unless YOU ARE SURE you want to purge all weather data - it will not be"
    echo "    recoverable after this action!"
    echo ""
    echo "    IF THAT IS NOT WHAT YOU WANT, HIT CTRL-C NOW TO ABORT THIS SCRIPT!"
    echo ""
    echo "################################################################################"
    echo ""
    echo -n "Pausing 30 seconds - CTRL-C to ABORT"
    for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    do
        sleep 1
        echo -n "."
    done
    echo ""
    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!" 
    echo "Are you sure you want to DESTROY all of your collected weather data?"
    echo -n "(NO/yes): "
    read INVAL
    if [ "yes" != "$INVAL" ]; then
        echo "Aborting database purge operation."
        exit 0
    else
        echo "Purging all weather data!"
    fi
}


purge_archive_data()
{
	echo "delete from archive where dateTime >= '0';" >> $WVIEW_DATA_DIR/commands.sql
	echo ".read $WVIEW_DATA_DIR/commands.sql" | sqlite3 $WVIEW_DATA_DIR/archive/wview-archive.sdb
	rm -rf $WVIEW_DATA_DIR/commands.sql
}


################################################################################
##################  S C R I P T  E X E C U T I O N  S T A R T  #################
################################################################################

# First test to make sure that wview is not running:
if [ -f $WVIEWD_PID ]; then
    echo "wviewd is running - stop wview before running this script..."
    exit 3
fi

# Make sure that the archive DB is there:
if [ ! -f $WVIEW_DATA_DIR/archive/wview-archive.sdb ]; then
    echo "wview archive database NOT FOUND"
    exit 4
fi

## Intro:
interactive_intro

## Purge all archive records:
echo "Deleting all archive records:"
purge_archive_data

## Delete dependent databases:
echo "Deleting auto-generated databases:"
rm -rf $WVIEW_DATA_DIR/archive/wview-hilow.sdb
rm -rf $WVIEW_DATA_DIR/archive/wview-history.sdb
rm -rf $WVIEW_DATA_DIR/archive/wview-noaa.sdb

echo ""
echo "################################################################################"
echo "--> ALL wview weather data purged!"
echo "################################################################################"

exit 0

